ambry-store/src/main/java/com.github.ambry.store/IndexSegment.java

```diff
@@ -211,13 +207,15 @@
             throw e;
           }
         }
+        if (immutable) {
+          index = Collections.unmodifiableNavigableMap(index);
```

Comment:
Just curious why you don't do this earlier on line 192.

Reply:
The `readFromFile()` method adds elements to the index. After the addition, the index will be immutable